shanoir-ng-studies/src/main/java/org/shanoir/ng/study/service/StudyServiceImpl.java

```diff
@@ -583,7 +585,15 @@ public List<Study> findChallenges() {
 
 	@Override
 	public List<Study> findPublicStudies() {
-		return this.studyRepository.findByVisibleByDefaultTrue();
+		List<Study> studyList = this.studyRepository.findByVisibleByDefaultTrue();
+		for (Study study : studyList) {
+			List<String> studyFlagList = new ArrayList<>();
+			String datasetType = (String) this.rabbitTemplate.convertSendAndReceive(RabbitMQConfiguration.STUDY_DATASET_TYPE, study.getId());
```

Comment:
You should get a list of string here

Reply:
All of this functionality (getting modalities of studies) will be done differently in another issue